### PR TITLE
Game logic

### PR DESCRIPTION
**Pull request by 1v4n4**
For this milestone, in a branch name “game_logic” we created the logic for our game.  File main.br in bin folder is responsible for user interaction, which prompts for users and collects user inputs. We created two classes, Board and Player and we covered edge cases if the player tries to do to cheat the game or perform invalid action. 
Tic-tac-toe is a simple game with two players, who are choosing X and 0 symbols until one of them makes a row, column, or a diagonal of the same symbols or all of the cells are occupied.

### CODE REVIEW

File: bin/main.rb [hunk 1]
```diff
@@ -1,107 +1,79 @@
...
-  [6, 7, 8], # 3rd horizontal row
-  [0, 3, 6], # 1st vertial column
-  [1, 4, 7], # 2nd vertial column
-  [2, 5, 8], # 3rd vertial column
-  [6, 4, 2], # right diagonal
-  [0, 4, 8] # left diagonal
-].freeze
+require './lib/game_logic'
+require './lib/helpers'
+require './lib/player'
```
Comment (eliasecasta) [inline bin/main.rb]: Please change for require_relative

**Review by eliasecasta — CHANGES_REQUESTED** [review]
# Status:  Changes Required ♻️

Hello @1v4n4 and @vikitaotiz,

You have done a great job with your game so far, please fix this issues before requesting for another code review please:

- [x] No `puts/print` and `gets.chomp` outside `main.rb` file. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/only_main_print.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))
- [x]  Please change the requires that are using your own files for require_relative like so: 
  
    require_relative '../lib/game_logic' (I was unable to run your app until I did this change)

OPTIONAL:

- [x] Use attr_readers/attr_writers unless you really need attr_accessors. You must be aware that by using attr_accessors you allow writing that property from outside the class. That variable can be set to anything.  ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/attr_accessor.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))


It was not required for this milestone but good job with your readme file, you are just missing the instructions to execute your app, you are just telling the user to cd to the TicTac folder and that is it.

Please correct those issues and send another review request, good luck, and well done!

File: bin/main.rb (hunk 1 above)
Comment (1v4n4) [reply bin/main.rb]: Done!

**Review by smcommits — COMMENTED** [review]
# Status: Invalid 🛑 

## Hello, team! Thank you so much for all the hard work. Your efforts are highly appreciated. Unfortunately, you have submitted a CRR for the Milestone - Instructions for Players but you have provided the PR link for Milestone - Game Logic. This CRR will be marked as invalid in your dashboard. Please request a new Code Review for the appropriate milestone.

**Review by smcommits — CHANGES_REQUESTED** [review]
# Status: Changes Requested 👍🏽 

## Hello, team! You have done a great job. Excellent work implementing the changes that the previous TSE requested. Unfortunately, the game keeps crashing while I am trying to play. 

## Required Changes 

- [x] **As soon as I am selecting a position, the game is crashing. Please fix this.**

GIF of the Game Crashing
:----------------------------------:
Peek 2021-05-08 00-48

As per my assumption, the reason this is happening is that you are trying to use a method that you haven't defined in your class interface. Also, if you are trying to access some instance variable, make sure you are using attr_reader. 

# 

#### Please reach out to me on Slack if you need assistance or believe that I have overlooked something and made an unfair request.

File: lib/game_logic.rb [hunk 2]
```diff
@@ -1,4 +1,6 @@
 module TicTacToe
+  attr_reader :board
```
Comment (Dipeshtwis) [inline lib/game_logic.rb:2]: Looks good now.

**Review by Dipeshtwis — APPROVED** [review]
# Status: Approved ✔️

## Hi @1v4n4 &  @vikitaotiz

Good job on implementing all the suggested changes asked in the previous review. You game is working fine.

**You may close this Pull Request and merge the changes**
  
**Keep being awesome devs and good luck with your next milestone!**
---------

Cheers and Happy coding!👏👏👏

Code review by **Dipesh Kumar**
(Feel free to contact me on slack by the handle of  **deep** if you have any additional questions).
[Dipesh Kumar](https://deeptwist.tk) | [Twitter](https://twitter.com/97deepeshkumar)

**Pull request merged**